src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java

```diff
@@ -0,0 +1,277 @@
+    {
+        return incomingEdge ->
+        {
+            if (!incomingEdge.getName().isPresent())
+            {
+                return false;
+            }
+            final String startEdgeName = startEdge.getName().get();
+            final String incomingEdgeName = incomingEdge.getName().get();
+            return checkBeforeEditDistance(incomingEdgeName, startEdgeName);
+        };
+    }
+
+    /**
+     * Wrapper for editDistanceIsOne(). Handles cases where the incomingEdge has the same name as
+     * the startingEdge and where the edit distance between the two names is greater than one before
+     * analyzing the edit distance.
+     *
+     * @param incomingEdgeName
+     *            the next edge in the search area
+     * @param startEdgeName
+     *            the edge from which the search started
+     * @return true if the edit distance between two edge's names is 1; false otherwise
+     */
+    private static boolean checkBeforeEditDistance(final String incomingEdgeName,
+            final String startEdgeName)
+    {
+        if (Math.abs(incomingEdgeName.length() - startEdgeName.length()) > 1)
+        {
+            return false;
+        }
+        if (incomingEdgeName.equals(startEdgeName))
+        {
+            return false;
+        }
+        return editDistanceIsOne(incomingEdgeName, startEdgeName);
+    }
+
+    /**
+     * The function used to collect {@link Edge}s that fall within the search area.
+     *
+     * @param startEdge
+     *            the edge from which the search started
+     * @param maximumSearchDistance
+     *            the maximum distance from the end of the incoming edge to the start of the
+     *            starting edge
+     * @return A stream of edges that fall in the search area
+     */
+    private static Function<Edge, Stream<Edge>> edgesWithinMaximumSearchDistance(
+            final Edge startEdge, final Distance maximumSearchDistance)
+    {
+        return incomingEdge -> incomingEdge.end().getLocation()
+                .distanceTo(startEdge.start().getLocation())
+                .isLessThanOrEqualTo(maximumSearchDistance)
+                        ? incomingEdge.connectedEdges().stream().filter(Edge::isMasterEdge)
+                        : Stream.empty();
+    }
+
+    /**
+     * Check the road names for identifier substring differences. If they have any return false. If
+     * they have none, we compute the edit distance in editDistanceIsOneHelper. This method should
+     * only take in Strings that are the same length, or whose lengths are off by a single
+     * character.
+     *
+     * @param incomingEdgeName
+     *            the name of the next edge in the search area
+     * @param startingEdgeName
+     *            the name of the edge from which the search started
+     * @return true if the edit distance between two edge's names is 1; false otherwise
+     */
+    private static boolean editDistanceIsOne(final String incomingEdgeName,
```

Comment:
This method, and the subsequent `editDistanceIsOneHelper()`, have confusing names. I would consider renaming this method to make it more clear that it checks for identifier substrings, and then maybe rename `editDistanceIsOneHelper()` to `editDistanceIsOne()` since it's the method that actually checks the edit distance as normally understood.